enricher/api/src/main/java/io/fabric8/maven/enricher/api/BaseEnricher.java

```diff
@@ -114,4 +127,32 @@ protected boolean isOpenShiftMode() {
         }
         return false;
     }
+
+    protected void ensureMetadata(PodTemplateSpecBuilder obj) {
+        ObjectMeta metadata = obj.getMetadata();
+        if (metadata == null) {
+            obj.withNewMetadata().endMetadata();
+        }
+    }
+
+    protected void addInitContainer(ObjectMetaBuilder metadata, JSONObject initContainer) {
+        String initContainerAnnotation = metadata.getAnnotations().get(INIT_CONTAINER_ANNOTATION);
```

Comment:
This is retrieving current annotation, hence `getAnnotations`, so that the two can be merged. Should always be non-null.